### PR TITLE
Do not use `FormattedModelValue` in password editor template

### PR DESCRIPTION
**Pull request by dougbu**
- #7418
- add quirk switch to reverse this if necessary

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/DefaultEditorTemplatesTest.cs [hunk 1]
```diff
@@ -24,6 +24,11 @@ namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
 {
     public class DefaultEditorTemplatesTest
     {
+        static DefaultEditorTemplatesTest()
+        {
+            AppContext.SetSwitch("Switch.Microsoft.AspNetCore.Mvc.UsePasswordValue", true);
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/DefaultEditorTemplatesTest.cs]: This was for local testing. I'll back the static constructor out.

File: test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/DefaultEditorTemplatesTest.cs [hunk 2]
```diff
@@ -521,6 +521,113 @@ public void MultilineTextTemplate_ReturnsTextArea()
...
+
+            // Act
+            var result = DefaultEditorTemplates.PasswordTemplate(helper);
+
+            // Assert
+            Assert.Equal(expected, HtmlContentUtilities.HtmlContentToString(result));
+        }
+
+        [Fact]
+        public void PasswordTemplate_ReturnsInputElement_UsesHtmlAttributes()
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/DefaultEditorTemplatesTest.cs]: This is really the only test you need. The others aren't directly related to this issue are they?

Comment (dougbu) [reply test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/DefaultEditorTemplatesTest.cs]: FYI, I consolidated the other tests into `PasswordTemplate_ReturnsInputElement_IgnoresValues()`, setting everything in that. Mixing this one into that test case could be fooled by precedence between the value sources.

**Comment by dougbu** [comment]
f061d328d9

**Pull request closed without merging**